src/Components/Input.tsx

```diff
@@ -75,28 +73,55 @@ class Input extends React.Component<InputProps, InputState> {
   }
 
   render() {
-    if (this.props.rightView) {
+    if (this.props.leftView && this.props.rightView) {
+      const newProps: any = { ...this.props }
+      delete newProps.className
```

Comment:
Rather than duplicating all of the props and then deleting, you can also omit:

```js
const { className, ...newProps } = this.props
```

Reply:
So in the above code `className` get restructured in its own thing and everything else is on `...newProps`?

Reply:
Aye 👍

Reply:
When I try this method of accessing the `...newProps` I'm getting a type error: 

Type '{ onFocus: any; onBlur: any; value: string | number | string[]; children?: ReactNode; error?: boo...' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<ThemedOuterStyledProps<ChangeTargetHTMLP...'.

Reply:
Oh I guess that’s because `newProps` is no longer cast as `any`. Try:

```ts
const { className, ...newProps }: any = this.props
```

Reply:
doesn't `any` drop all the type goodness? I can't explain why but [dropping `ref`](https://github.com/artsy/reaction/blob/77a496f/src/Components/CircleIcon.tsx#L22) may make the editor happy:

```ts
const { ref, ...reminderProps } = props
```

Reply:
Yes it would, good call 👍

Reply:
Sweet!